src/main/java/io/vertx/json/schema/JsonSchemaOptions.java

```diff
@@ -40,6 +41,11 @@ public class JsonSchemaOptions {
    */
   private OutputFormat outputFormat = OutputFormat.Flag;
 
+  /**
+   * The json format validator that is used during schema validation.
+   */
+  private JsonFormatValidator jsonFormatValidator = new DefaultJsonFormatValidatorImpl();
```

Comment:
Now when I think about this, it is might be a problem, because JsonSchemaOptions is annotated with JsonGen and we can't serialize the JsonFormatValidator.

And using `@GenIgnore` would make this unusable in Kotlin, etc .. If we cannot find a better solution, we could store the FQN and instantiate the class during deserialization. I have to double check with @vietj .

Reply:
Okay, I talked to Julien. He made me aware of the builder pattern guidelines [1]. So we should not mix logic or algorithm with a DataObject. This means that we need an additional builder for SchemaRepository and Validator.

I will try to prepare a proposal this week.

[1] https://github.com/eclipse-vertx/vert.x/wiki/Builder-pattern-guidelines